airflow/www/static/js/dag/details/dagRun/MarkRunAs.tsx

```diff
@@ -58,42 +65,115 @@ const MarkRunAs = ({ runId, state, ...otherProps }: Props) => {
     markSuccess({ confirmed: true });
   };
 
+  useEffect(() => {
+    const storedSuccessValue = localStorage.getItem("doNotShowAgainSuccess");
+    const storedFailedValue = localStorage.getItem("doNotShowAgainFailed");
+
+    if (storedSuccessValue) {
+      setDoNotShowAgainSuccess(JSON.parse(storedSuccessValue));
+    }
+    if (storedFailedValue) {
+      setDoNotShowAgainFailed(JSON.parse(storedFailedValue));
+    }
+  }, []);
+
+  const confirmAction = () => {
+    localStorage.setItem(
+      "doNotShowAgainSuccess",
+      JSON.stringify(doNotShowAgainSuccess)
+    );
+    localStorage.setItem(
+      "doNotShowAgainFailed",
+      JSON.stringify(doNotShowAgainFailed)
+    );
+
+    if (confirmingAction === "failed") {
+      markAsFailed();
+    } else if (confirmingAction === "success") {
+      markAsSuccess();
+    }
+    setShowConfirmationModal(false);
+  };
+
   useKeysPress(keyboardShortcutIdentifier.dagMarkSuccess, () => {
-    if (state !== "success") markAsSuccess();
+    if (state !== "success") {
+      if (!doNotShowAgainSuccess) {
+        setConfirmingAction("success");
+        setShowConfirmationModal(true);
```

Comment:
When you have states that are bound to each other by some logic, I recommend using a useReducer. This way coupled logic between states get hidden behind a common action, and this would go into the reducer. 

https://react.dev/reference/react/useReducer

Action could for instance be `CONFIRM_SUCCESS`, `CONFIRM_FAILURE` etc. Then it would take care of switching all the appropriate states, and when you need it you can just dispatch the appropriate action.